java/code/src/com/redhat/rhn/frontend/action/systems/audit/ScapSetupAction.java

```diff
@@ -24,48 +23,67 @@
-    private static final String SCAP_ENABLED = "scapEnabled";
-    private static final String REQUIRED_PKG = "requiredPackage";
-    private static final String OPENSCAP_SUSE_PKG = "openscap-utils";
-    private static final String OPENSCAP_REDHAT_PKG = "openscap-scanner";
-    private static final String OPENSCAP_DEBIAN_PKG = "libopenscap8";
+    public static final String SCAP_ENABLED = "scapEnabled";
+    public static final String REQUIRED_PKG = "requiredPackage";
+
+    private static final List<String> OPENSCAP_SUSE_PKG = List.of("openscap-utils");
+    private static final List<String> OPENSCAP_REDHAT_PKG = List.of("openscap-scanner");
+    private static final List<String> OPENSCAP_DEBIAN_PKG = List.of("libopenscap25", "openscap-common");
+    private static final List<String> OPENSCAP_DEBIAN_LEGACY_PKG = List.of("libopenscap8");
 
     protected void setupScapEnablementInfo(RequestContext context) {
         Server server = context.lookupAndBindServer();
         User user = context.getCurrentUser();
-        boolean enabled = false;
-        String requiredPkg = "";
+
+        boolean enabled;
+        List<String> requiredPackages;
+
         if (server.asMinionServer().isPresent()) {
             MinionServer minion = server.asMinionServer().get();
             switch (minion.getOsFamily()) {
                 case "Suse":
-                    requiredPkg = OPENSCAP_SUSE_PKG;
+                    requiredPackages = OPENSCAP_SUSE_PKG;
                     break;
 
                 case "Debian":
-                    requiredPkg = OPENSCAP_DEBIAN_PKG;
+                    requiredPackages = getPackagesForVersion(minion.getRelease());
                     break;
 
                 default:
-                    requiredPkg = OPENSCAP_REDHAT_PKG;
-            }
-            InstalledPackage installedPkg =
-                    PackageFactory.lookupByNameAndServer(requiredPkg, server);
-            if (installedPkg != null) {
-                enabled = true;
+                    requiredPackages = OPENSCAP_REDHAT_PKG;
             }
+
+            // Verify all the packages are installed
+            enabled = requiredPackages.stream()
+                .map(pkg -> PackageFactory.lookupByNameAndServer(pkg, server))
+                .noneMatch(Objects::isNull);
         }
         else {
             enabled = ScapManager.isScapEnabled(server, user);
+            requiredPackages = List.of();
         }
+
         context.getRequest().setAttribute(SCAP_ENABLED, enabled);
-        context.getRequest().setAttribute(REQUIRED_PKG, requiredPkg);
+        context.getRequest().setAttribute(REQUIRED_PKG, String.join(" ", requiredPackages));
 
         SdcHelper.ssmCheck(context.getRequest(), server.getId(), user);
     }
+
+    private static List<String> getPackagesForVersion(String debianRelease) {
+        if (NumberUtils.isParsable(debianRelease) && Integer.parseInt(debianRelease) <= 11) {
```

Comment:
This can result in a NumberFormatException when an Ubuntu version is passed

```
Caused by: java.lang.NumberFormatException: For input string: "22.04"
        at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.base/java.lang.Integer.parseInt(Integer.java:652)
        at java.base/java.lang.Integer.parseInt(Integer.java:770)
        at com.redhat.rhn.frontend.action.systems.audit.ScapSetupAction.getPackagesForVersion(ScapSetupAction.java:84)
        at com.redhat.rhn.frontend.action.systems.audit.ScapSetupAction.setupScapEnablementInfo(ScapSetupAction.java:60)
        at com.redhat.rhn.frontend.action.systems.audit.ListScapAction.execute(ListScapAction.java:64)
        at org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:431)
```

Reply:
I am not sure why, but 4.3 testsuite show this error. Could be that Ubuntu report also OsFamily() "Debian"

Reply:
OsFamily is the grain value and here Ubuntu is Debian. You can use getOs() to differ between Ubuntu and Debian

Reply:
Ouch... I thought this was only for Debian and I haven't tested Ubuntu. At this point, I'm also not sure if the package is correct for Ubuntu and which version needs what. I will address this on Monday.

Reply:
I can confirm this. We see this exact error in the Tomcat logs in the 4.3 CI and get an ISE when visiting the Audit page on a Ubuntu Minion.